src/components/TimePicker/TimePicker.js

```diff
@@ -87,15 +86,19 @@ function TimePicker({forwardedRef, defaultValue, onSubmit, onInputChange}) {
     const [selectionHour, setSelectionHour] = useState({start: 0, end: 0});
     const [selectionMinute, setSelectionMinute] = useState({start: 2, end: 2}); // we focus it by default so need  to have selection on the end
     const [hours, setHours] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).hour);
-    const [minute, setMinute] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).minute);
+    const [minutes, setMinutes] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).minute);
     const [amPmValue, setAmPmValue] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).period);
+    const [lastPressedKey, setLastPressedKey] = useState('');
 
     const hourInputRef = useRef(null);
     const minuteInputRef = useRef(null);
 
     const focusMinuteInputOnFirstCharacter = useCallback(() => {
-        const cleanupTimer = setSelection({start: 0, end: 0}, minuteInputRef, setSelectionMinute);
-        return cleanupTimer;
+        setSelectionMinute({start: 0, end: 0});
+        const timer = setTimeout(() => {
+            minuteInputRef.current.focus();
+        }, 10);
+        return () => clearTimeout(timer);
```

Comment:
setTimeout is discouraged

Reply:
This is how it's already implemented in `main` for hours. I just fixed it for minutes.

https://github.com/Expensify/App/blob/88d837c2e88c8c4c22a4c80f93b622d634e67290/src/components/TimePicker/TimePicker.js#L100-L106

Reply:
I think we should remove both the `setTimeout`(s) then. The use of setTimeout here is hacky

Reply:
In such a case, the setSelection will not work before the focus. Could you suggest a solution for this without using the timeout?

Reply:
Can you specify the steps to reproduce this?

Reply:
If you remove the timers and try moving from one input to another using arrow keys, you'll see that the inputs are getting focused in-between the digits for some reason:

https://github.com/Expensify/App/assets/12595293/8192825d-1125-46d8-9b16-85447d1adca8

Reply:
I see. Here we are moving the cursor twice, the first one programatically `setSelectionMinute` and the second one is by the arrow key event itself. To fix this we need to prevent the later. In `arrowLeftCallback` / `arrowRightCallback` call `e.preventDefault()` before focusing the minutes

Reply:
Yeah, that's a great catch, thanks!

Just a note: there was an attempt in the original PR to handle the incorrect focus on iOS by using an ios-specific file: https://github.com/Expensify/App/commit/91b8be1e7f91e3a6df4f2ee88cd4705d982d6f42#diff-c01367cc0050be19460e165b444bf846d459db24beb5f8e0620b9146c3c7c7fc

I tried to simplify the code by moving everything inside the original `TimePicker` file under a reusable function: `setCursorPosition`. Please check if it's an acceptable change from your POV. It doesn't harm non-ios platforms, but does the job for iOS by calling `ref.current?.setNativeProps({selection})`

Reply:
Also, please disregard the lint issues for now – I'm still about to work on the `lastPressedKey` thing.

Reply:
Why is the setSelection still using timeouts?

Reply:
Could you please point me to the code?

Reply:
It's reverted now.

Can we remove the platform dependent code now?

Reply:
> It's reverted now.

It didn't use timeouts 🙂 
https://github.com/Expensify/App/pull/33344/commits/367f9f1dd5f77f1070f25495eb205fc3e8ca96aa#diff-e416eb0ab69accefaf15924d73febbc248ba3ed817d6ee2539be167b9cead60e

Reply:
> Can we remove the platform dependent code now?

Unfortunately no, for some reason when I tested it the first time, the code was cached and worked well on the Web.
After a re-test, it started crashing with the `setNativeProps is not a function` error.
Therefore, we need to keep the platform-dependent code because iOS needs `setNativeProps`.

Reply:
Can you list the steps to reproduce the bug if we don't use setNativeProps on iOS?

Reply:
Just casually type the hours to make the cursor move to the minutes. The cursor will appear at the end, but once you continue entering, the cursor will jump to the start of the minutes:


https://github.com/Expensify/App/assets/12595293/ddca3058-651a-49d1-9f00-9db83931dd04